src/adapter/src/error.rs

```diff
@@ -306,6 +311,25 @@ impl AdapterError {
             AdapterError::UnallowedOnCluster { cluster, .. } => (cluster == MZ_INTROSPECTION_CLUSTER.name).then(||
                 format!("The transaction is executing on the {cluster} cluster, maybe having been routed there by the first statement in the transaction.")
             ),
+            AdapterError::InputNotReadableAtRefreshAtTime(oracle_read_ts, earliest_possible) => {
+                Some(format!(
+                    "The requested REFRESH AT time is {}, \
+                    while the following input collections are readable at no earlier than the following times: [{}].",
+                    oracle_read_ts,
+                    earliest_possible.iter().map(|(antichain, id_bundle)|
+                        format!(
+                            "[{}]: {}",
+                            id_bundle.iter().map(|id| format!("{}", id)).join(", "),
+                            if antichain.len() == 1 {
+                                format!("{}", antichain.as_option().expect("antichain contains exactly 1 timestamp"))
+                            } else {
+                                // This can't occur currently
+                                format!("{:?}", antichain)
+                            }
+                        )
+                    ).join("; "),
+                ))
+            }
```

Comment:
Uncovered according to [coverage report.](https://buildkite.com/materialize/coverage/builds/356)

Reply:
Actually, there is a test that provokes this error, but the `detail()` part of errors are not printed in slts, so that's why this part of the error msg is not covered. [But I tested this manually.](https://github.com/MaterializeInc/materialize/pull/25184#discussion_r1489286312)

This is the test:
https://github.com/MaterializeInc/materialize/blob/0f35ab4e2ca3a526e81d472f87f3773e9acd7c3a/test/sqllogictest/materialized_views.slt#L978